src/Symfony/Component/Clock/Tests/ClockTest.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Clock\Tests;
+
+use PHPUnit\Framework\TestCase;
+use Symfony\Component\Clock\Clock;
+use Symfony\Component\Clock\MockClock;
+use Symfony\Component\Clock\NativeClock;
+
+class ClockTest extends TestCase
+{
+    public function testClock()
+    {
+        $this->assertInstanceOf(\DateTimeImmutable::class, Clock::now());
```

Comment:
Isn't this guaranteed by the compiler, as the method has return type set to `\DateTimeImmutable`?

Reply:
we ensure an immutable variant explicitly this way